motionplan/ptgGroupFrame.go

```diff
@@ -0,0 +1,193 @@
+package motionplan
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/edaniels/golog"
+	pb "go.viam.com/api/component/arm/v1"
+
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+const (
+	ptgIndex int = iota
+	trajectoryAlphaWithinPTG
+	distanceAlongTrajectoryIndex
+)
+
+type ptgFactory func(float64, float64) tpspace.PrecomputePTG
+
+var defaultPTGs = []ptgFactory{
+	tpspace.NewCirclePTG,
+	tpspace.NewCCPTG,
+	tpspace.NewCCSPTG,
+	tpspace.NewCSPTG,
+	tpspace.NewSideSPTG,
+	tpspace.NewSideSOverturnPTG,
+}
+
+type ptgGroupFrame struct {
+	name          string
+	limits        []referenceframe.Limit
+	geometries    []spatialmath.Geometry
+	ptgs          []tpspace.PTG
+	velocityMMps  float64
+	turnRadMeters float64
+	logger        golog.Logger
+}
+
+// NewPTGFrameFromTurningRadius will create a new Frame which is also a PTGProvider. It will precompute the default set of
+// trajectories out to a given distance, or a default distance if the given distance is <= 0.
+func NewPTGFrameFromTurningRadius(
+	name string,
+	logger golog.Logger,
+	velocityMMps, turnRadMeters, refDist float64,
+	geoms []spatialmath.Geometry,
+) (referenceframe.Frame, error) {
+	if velocityMMps <= 0 {
+		return nil, fmt.Errorf("cannot create ptg frame, movement velocity %f must be >0", velocityMMps)
+	}
+	if turnRadMeters <= 0 {
+		return nil, fmt.Errorf("cannot create ptg frame, turning radius %f must be >0", turnRadMeters)
+	}
+
+	if refDist <= 0 {
+		// Default to a distance of just over one half of a circle turning at max radius
+		refDist = 1000. * turnRadMeters * math.Pi * 0.9
+	}
+
+	// Get max angular velocity in radians per second
+	maxRPS := velocityMMps / (1000. * turnRadMeters)
+	pf := &ptgGroupFrame{name: name}
+	err := pf.initPTGs(logger, velocityMMps, maxRPS, refDist)
+	if err != nil {
+		return nil, err
+	}
+
+	pf.geometries = geoms
+	pf.velocityMMps = velocityMMps
+	pf.turnRadMeters = turnRadMeters
+
+	pf.limits = []referenceframe.Limit{
+		{Min: 0, Max: float64(len(pf.ptgs) - 1)},
+		{Min: -math.Pi, Max: math.Pi},
+		{Min: 0, Max: refDist},
+	}
+
+	return pf, nil
+}
+
+// newPTGFrameFromPTGFrame will create a new Frame from a preexisting ptgGroupFrame, allowing the adjustment of `refDist` while keeping
+// other params the same. This may be expanded to allow altering turning radius, geometries, etc.
+func newPTGFrameFromPTGFrame(frame referenceframe.Frame, refDist float64) (referenceframe.Frame, error) {
```

Comment:
Yeah I'm fine with it, but it just feels like the function you want here is "change refDist but also make a new object when doing it" and this name doesn't communicate that very well.  Not a big deal, and if we end up needing to expand on this ever we can reconsider it

Reply:
> if we end up needing to expand on this ever we can reconsider it

That was the main thought I had behind building it this way rather than pushing off the `refDist` change to some internal method that was called differently; if we want to change other things in the future that will slot nicely in here.